packages/@aws-cdk/aws-glue/lib/database.ts

```diff
@@ -41,6 +56,13 @@ export interface DatabaseProps {
    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-database-databaseinput.html
    */
   readonly locationUri?: string;
+
+  /**
+   * The target database for resource linking.
+   *
+   * @default undefined. The database will not be created as a resource link.
+   */
+  readonly targetDatabase?: TargetDatabase;
```

Comment:
Would it make sense to type this property with `IDatabase` instead to provide more type safety? Would that incur some limitation?